src/api/v2/types.rs

```diff
@@ -80,6 +81,24 @@ pub struct Status {
 	pub partition: Option<Partition>,
 }
 
+#[derive(Deserialize)]
+#[serde(rename_all = "kebab-case")]
+pub enum Transaction {
+	Data(String),
+	Extrinsic(String),
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct TransactionHash {
+	pub hash: String,
```

Comment:
You can use `sp_core::Hash` directly here. It implements serialize and that way it will be deserialized smth like this:
```
{"hash":"0x0000000000000000000000000000000000000000000000000000000000000000"}
```

Reply:
I would personally just remove this structure at all and remove just `warp::reply::Json<sp_core::H256>` directly from `submit` method

Reply:
Still unsure if we want `0x` prefix or not, but Ill leave it with prefix as it si simpler atm, and get back to this later. At this point there will be more fields in respose (which can change in the future), so second comment doesn't apply.

Reply:
I think it doesn't matter if there will be a prefix. So for ease lets keep it.

As for second comment. The structure is called `TransactionHash` which is really weird, so if we have plans to add fields lets call it `SubmitResponse` or smth

Reply:
Sorry, didn't saw last force push. All good!